workflow/cron/operator.go

```diff
@@ -108,14 +109,25 @@ func getWorkflowObjectReference(wf *v1alpha1.Workflow, runWf *v1alpha1.Workflow)
 }
 
 func (woc *cronWfOperationCtx) persistUpdate() {
-	data, err := json.Marshal(map[string]interface{}{"status": woc.cronWf.Status})
+	woc.patch(map[string]interface{}{"status": woc.cronWf.Status})
+}
+
+func (woc *cronWfOperationCtx) persistUpdateActiveWorkflows() {
+	woc.patch(map[string]interface{}{"status": map[string]interface{}{"active": woc.cronWf.Status.Active}})
+}
+
+func (woc *cronWfOperationCtx) patch(patch map[string]interface{}) {
+	data, err := json.Marshal(patch)
 	if err != nil {
-		woc.log.WithError(err).Error("failed to marshall cron workflow status data")
+		woc.log.WithError(err).Error("failed to marshall cron workflow status.active data")
 		return
 	}
 	err = wait.ExponentialBackoff(retry.DefaultBackoff, func() (bool, error) {
 		cronWf, err := woc.cronWfIf.Patch(woc.cronWf.Name, types.MergePatchType, data)
 		if err != nil {
+			if argoerr.IsTransientErr(err) {
```

Comment:
I think this logic is wrong - I think we, should be `!argoerr.IsTransientErr(err)`

Reply:
I actually think this is correct. If the error is transient we should retry the operation. `ExponentialBackoff` will only do so if we return `false, nil`, otherwise the retry mechanism will stop.

You can see an existing example here:

https://github.com/argoproj/argo/blob/b1e2c207722be8ec9f26011957ccdeaa95da2ded/workflow/controller/operator.go#L3026-L3035

In this case if the error is _not_ transient, we return with `err`. If it is, we return with `nil`